node/worker/worker.go

```diff
@@ -288,39 +288,65 @@ func (w *Worker) SuperCommitteeForNextEpoch(
 ) (shard.State, error) {
 	var (
 		nextCommittee shard.State
-		oops          error
+		err           error
 	)
 	switch shardID {
 	case shard.BeaconChainShardID:
 		if shard.Schedule.IsLastBlock(w.current.header.Number().Uint64()) {
-			nextCommittee, oops = committee.WithStakingEnabled.Compute(
+			nextCommittee, err = committee.WithStakingEnabled.Compute(
 				new(big.Int).Add(w.current.header.Epoch(), common.Big1),
-				w.config,
 				beacon,
 			)
 		}
 	default:
-		// WARN When we first enable staking, this condition may not be robust by itself.
-
+		// TODO: needs to make sure beacon chain sync works.
+		beaconEpoch := beacon.CurrentHeader().Epoch()
 		if w.config.IsStaking(w.current.header.Epoch()) {
-			switch beacon.CurrentHeader().Epoch().Cmp(w.current.header.Epoch()) {
+			switch beaconEpoch.Cmp(w.current.header.Epoch()) {
 			case 1:
-				nextCommittee, oops = committee.WithStakingEnabled.ReadFromDB(
-					beacon.CurrentHeader().Epoch(), beacon,
+				// If beacon chain is bigger than shard chain in epoch, it means I should catch up with beacon chain now
+				nextCommittee, err = committee.WithStakingEnabled.ReadFromDB(
+					beaconEpoch, beacon,
 				)
+				blockEpoch := big.NewInt(0).Set(beaconEpoch).Sub(beaconEpoch, big.NewInt(1)) // Set this block's epoch to be beaconEpoch - 1, so the next block will have beaconEpoch
```

Comment:
What is wrong about setting current epoch to beaconEpoch?

Reply:
It's already explicit. beaconEpoch - 1 so the next block will have beaconEpoch itself. If we set current block to be beaconEpoch, then next block will have beaconEpoch + 1, which is wrong. The logic is whenever there is a block with Shard State in it, the next block is the new epoch and that epoch is preBlock.Epoch + 1.

Reply:
Thank you, I mean put that directly into a longer /* */, because valuable information for programmer